internals/cli/cmd_run_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2024 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cli_test
+
+import (
+	"io/fs"
+	"os"
+	"path"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/canonical/pebble/internals/cli"
+)
+
+func (s *PebbleSuite) TestMaybeCopyPebbleDir(c *C) {
+	src := c.MkDir()
+	err := os.MkdirAll(path.Join(src, "a", "b", "c"), 0700)
+	c.Assert(err, IsNil)
+	err = os.WriteFile(path.Join(src, "a", "b", "c", "a.yaml"), []byte("# hi\n"), 0666)
+	c.Assert(err, IsNil)
+	err = os.WriteFile(path.Join(src, "a.yaml"), []byte("# bye\n"), 0666)
+	c.Assert(err, IsNil)
+
+	dst := c.MkDir()
+	err = cli.MaybeCopyPebbleDir(dst, src)
+	c.Assert(err, IsNil)
+
+	ok := 0
+	dstFS := os.DirFS(dst)
+	err = fs.WalkDir(dstFS, ".", func(p string, d fs.DirEntry, err error) error {
```

Comment:
Sure thing.

Reply:
As Harry mentioned on chat, "fs.WalkDir is so much better because all the paths are rooted at the directory used to create the fs" and there's no need for a bunch of `filepath.Rel`. I'd missed this, sorry!